src/OpenTK/Platform/MacOS/CocoaContext.cs

```diff
@@ -46,17 +46,52 @@ internal class CocoaContext : DesktopGraphicsContext
         private static readonly IntPtr selMakeCurrentContext = Selector.Get("makeCurrentContext");
         private static readonly IntPtr selUpdate = Selector.Get("update");
 
-        private static readonly IntPtr opengl = NS.AddImage(
+        private static IntPtr opengl = NS.AddImage(
             "/System/Library/Frameworks/OpenGL.framework/OpenGL",
             AddImageFlags.ReturnOnError);
 
-        private static readonly IntPtr opengles = NS.AddImage(
+        private static IntPtr opengles = NS.AddImage(
             "/System/Library/Frameworks/OpenGL.framework/OpenGLES",
             AddImageFlags.ReturnOnError);
+        
+        private static bool bindUsingDlSym = false;
 
         static CocoaContext()
         {
             Cocoa.Initialize();
+            DetectBindingMethod();
+        }
+
+        // As of MacOS Monterey Beta 1/2, NSLookupSymbolInImage() stopped working. It is unclear
+        // if this will be fixed by Apple in the future. dlsym still works, but is about 20x slower.
+        // We will try to detect if NSLookup works by looking up 'glClear' and fallback to dlsym 
+        // if it returns NULL.
```

Comment:
It would be creat if we could add a link to #1308 in this comment so people can read more, also adding a date to the comment is also pretty useful.

Reply:
Sure i can do that.